workflows/analyze.smk

```diff
@@ -138,6 +155,18 @@ rule calculate_rhats_4chains:
         _, result4 = yg.serialize.read_metric_result(fp)
         result4 = np.array(result4)
 
+        # check the chains all have equal length
+        assert len(result1) == len(result2) == len(result3) == len(result4)
+        # thin the chains if they are longer than 10000 iterations, to avoid memory issues
+        # to about 10000 iterations
+        if len(result1) > 10000:
+            # get thinning factor
+            thinning_factor = int(len(result1) / 10000)
+            result1 = result1[::thinning_factor]
+            result2 = result2[::thinning_factor]
+            result3 = result3[::thinning_factor]
+            result4 = result4[::thinning_factor]
+
```

Comment:
Thinning the chains now for rhat. - as apparently even 20 GB of ram was not enough. - see suspected reason below.